hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/LegacyBucketHandler.java

```diff
@@ -212,13 +220,93 @@ public long handleDirectKeys(long parentId, boolean withReplica,
       return 0;
     }
 
+    if (enableFileSystemPaths) {
+      keyDataSizeWithReplica +=
+          handleDirectKeysForFSOLayout(parentId, withReplica, listFile, duData,
+              keyTable, seekPrefix, nsSummary);
+    } else {
+      keyDataSizeWithReplica +=
+          handleDirectKeysForOBSLayout(withReplica, listFile, duData, keyTable,
+              seekPrefix);
+    }
+
+    return keyDataSizeWithReplica;
+  }
+
+  /**
+   * This method handles disk usage of direct keys for OBS layout.
+   * @param withReplica if withReplica is enabled, set sizeWithReplica
+   * @param listFile if listFile is enabled, append key DU as a subpath
+   * @param duData the current DU data
+   * @param keyTable the key table
+   * @param seekPrefix the seek prefix used to position the iterator
+   * @return the total DU of all direct keys
+   * @throws IOException
+   */
+  public long handleDirectKeysForOBSLayout(boolean withReplica,
+                                           boolean listFile,
+                                           List<DUResponse.DiskUsage> duData,
+                                           Table<String, OmKeyInfo> keyTable,
+                                           String seekPrefix)
+      throws IOException {
+    long keyDataSizeWithReplica = 0L;
+
+
+    try (
+        TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>> iterator = keyTable.iterator()) {
+      iterator.seek(seekPrefix);
+
+      while (iterator.hasNext()) {
+        Table.KeyValue<String, OmKeyInfo> kv = iterator.next();
+        String dbKey = kv.getKey();
+
+        if (!dbKey.startsWith(seekPrefix)) {
+          break;
+        }
+
+        OmKeyInfo keyInfo = kv.getValue();
+        if (keyInfo != null) {
+          createDiskUsage(keyInfo, withReplica, listFile, duData);
+          if (withReplica) {
+            long keyDU = keyInfo.getReplicatedSize();
+            keyDataSizeWithReplica += keyDU;
+          }
+        }
+      }
+    }
+
+    return keyDataSizeWithReplica;
+  }
+
+  /**
+   * This method handles disk usage of direct keys for OBS layout.
+   * @param parentId parent directory/bucket
+   * @param withReplica if withReplica is enabled, set sizeWithReplica
+   * @param listFile if listFile is enabled, append key DU as a subpath
+   * @param duData the current DU data
+   * @param keyTable the key table
+   * @param seekPrefix the seek prefix used to position the iterator
+   * @param nsSummary of the parent directory/bucket
+   * @return the total DU of all direct keys
+   * @throws IOException
+   */
+  public long handleDirectKeysForFSOLayout(long parentId, boolean withReplica,
```

Comment:
This is not FSO, rename handleDirectKeysForLegacyFS, i.e. legacy file system and update comment also